elastic/kibana
# [Discover] Fix loading data views in ES|QL mode

- Closes https://github.com/elastic/kibana/issues/199065

## Summary

This PR makes sure that data views are loaded so user can view them after switching from ES|QL mode to data view mode. 

Looks like it was introduced in https://github.com/elastic/kibana/pull/195670

### Testing

On your local instance having `kibana_sample_data_logs` installed

* Open a link to an ES|QL query [link](http://localhost:5601/xot/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:60000),time:(from:now-15m,to:now))&_a=(columns:!(),dataSource:(dataViewId:e3465e67bdeced2befff9f9dca7ecf9c48504cad68a10efd881f4c7dd5ade28a,type:dataView),filters:!(),interval:auto,query:(language:kuery,query:''),sort:!(!(timestamp,desc))))
* Switch to classic
*  Open the DataView Picker, to see a list of DataViews, before this fix there were none

### Checklist
